spec/lang/step/intrinsics.md

```diff
@@ -383,3 +383,69 @@ impl<M: Memory> Machine<M> {
     }
 }
 ```
+
+We also implement the atomic fetch operations. First we define a helper function to decide which operations can be in a fetch operation.
+
+```rust
+/// Predicate to indicate if integer bin-op can be used for atomic fetch operations.
+fn is_atomic_binop(op: BinOpInt) -> bool {
```

Comment:
Why do we only support some of them?

Reply:
I only added support for the atomic fetch operations that are exposed by the Rust Atomics and by LLVM. My best guess is that since multiply and division are both cycle intensive operations they are not used atomically by the ISA.

But if you prefer having all integer operations as atomic, I don't have a problem with that.

Reply:
Hm okay that's fair, but needs a comment.